src/router/db.ts

```diff
@@ -100,24 +107,19 @@ db.post('/export', async (req: Request, res: Response) => {
   }
 });
 
-db.get('/list', async (req: Request, res: Response) => {
-  logger.info('Calling /list');
-  const uid = dbMan.getUid(req.auth);
-  const email = dbMan.getEmail(req.auth);
-  try {
-    const dbs = await MetadataRepo.getDbs(uid, email);
-    res.json(dbs);
-  } catch (e) {
-    res.status(500).end(logErrSentry(e, uid, email));
-  }
-});
-
-db.get('/disconnect/:dbAlias', async (req: Request, res: Response) => {
+db.post('/disconnect', async (req: Request, res: Response) => {
   logger.info('Calling /disconnect');
-  const { dbAlias } = req.params;
+  const { dbAlias, user } = req.body;
```

Comment:
User can send arbitrary post data to the engine, right? In that case I'm worried if this might go in a bad way and `iasql` user database will be dropped?

Reply:
Nope. Here's why:

1. The `pgsql_http` extension is not accessible to any account but the primary account, or functions defined by the primary account with the `security definer` configuration set.
2. The `http` endpoints they can call and the JSON blobs they create are fixed (or "quasi-static") no arbitrary, unsanitized inputs from the user are allowed.
3. The `user` property, in particular, is populated by the `session_user` global variable that the user can't touch. It is the username of the Postgres account calling that function, so they only way to change the `user` is to successfully log in as another user.